### PR TITLE
Replace `make_tokenize` with `make_tokenizer_located`

### PR DESCRIPTION
**Pull request by harupy**
This PR replaces `make_tokenize` with `make_tokenizer_located` to simplify the code.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -47,9 +47,9 @@ regex = { version = "1.6.0" }
 ropey = { version = "1.5.0", features = ["cr_lines", "simd"], default-features = false }
 ruff_macros = { version = "0.0.195", path = "ruff_macros" }
 rustc-hash = { version = "1.1.0" }
-rustpython-ast = { features = ["unparse"], git = "https://github.com/RustPython/RustPython.git", rev = "247e815880766d556ef1ca6f0af69daf1a5fe59a" }
-rustpython-common = { git = "https://github.com/RustPython/RustPython.git", rev = "247e815880766d556ef1ca6f0af69daf1a5fe59a" }
-rustpython-parser = { features = ["lalrpop"], git = "https://github.com/RustPython/RustPython.git", rev = "247e815880766d556ef1ca6f0af69daf1a5fe59a" }
+rustpython-ast = { features = ["unparse"], git = "https://github.com/RustPython/RustPython.git", rev = "68d26955b3e24198a150315e7959719b03709dee" }
```
Comment (harupy) [inline Cargo.toml:50]: https://github.com/RustPython/RustPython/pull/4369 fixed the issue where RustPython's lexer always starts at (1, 0).

**Comment by charliermarsh** [comment]
This is great!

**Comment by harupy** [comment]
@charliermarsh I found `locate_cmpops` also uses `make_tokenizer`. Feel free to fix this if you want to.

**Comment by charliermarsh** [comment]
I wonder if RustPython could leverage this to fix locations within f-strings.

**Pull request merged**